storm-core/src/jvm/org/apache/storm/scheduler/IsolationScheduler.java

```diff
@@ -0,0 +1,417 @@
+        }
+        cluster.setBlacklistedHosts(origBlacklist);
+    }
+
+    public Set<ExecutorDetails> removeElemFromSet(Set<Set<ExecutorDetails>> executorsSets) {
+        Set<ExecutorDetails> elem = executorsSets.iterator().next();
+        executorsSets.remove(elem);
+        return elem;
+    }
+
+    private List<TopologyDetails> isolatedTopologies(Collection<TopologyDetails> topologies) {
+        Set<String> topologyNames = isoMachines.keySet();
+        List<TopologyDetails> isoTopologies = new ArrayList<TopologyDetails>();
+        for (TopologyDetails topo : topologies) {
+            if (topologyNames.contains(topo.getName())) {
+                isoTopologies.add(topo);
+            }
+        }
+        return isoTopologies;
+    }
+
+    private Set<String> isolatedTopoplogyIds(List<TopologyDetails> topologies) {
+        Set<String> ids = new HashSet<String>();
+        if (topologies != null && topologies.size() > 0) {
+            for (TopologyDetails topology : topologies) {
+                ids.add(topology.getId());
+            }
+        }
+        return ids;
+    }
+
+    // map from topology id -> set of sets of executors
+    private Map<String, Set<Set<ExecutorDetails>>> topologyWorkerSpecs(List<TopologyDetails> topologies) {
+        Map<String, Set<Set<ExecutorDetails>>> workerSpecs = new HashMap<String, Set<Set<ExecutorDetails>>>();
+        for (TopologyDetails topology : topologies) {
+            workerSpecs.put(topology.getId(), computeWorkerSpecs(topology));
+        }
+        return workerSpecs;
+    }
+
+    private Map<String, List<AssignmentInfo>> hostAssignments(Cluster cluster) {
+        Collection<SchedulerAssignment> assignmentValues =  cluster.getAssignments().values();
+        Map<String, List<AssignmentInfo>> hostAssignments = new HashMap<String, List<AssignmentInfo>>();
+
+        for (SchedulerAssignment sa : assignmentValues) {
+            Map<WorkerSlot, List<ExecutorDetails>> slotExecutors = Utils.reverseMap(sa.getExecutorToSlot());
+            Set<Map.Entry<WorkerSlot, List<ExecutorDetails>>> entries = slotExecutors.entrySet();
+            for (Map.Entry<WorkerSlot, List<ExecutorDetails>> entry : entries) {
+                WorkerSlot slot = entry.getKey();
+                List<ExecutorDetails> executors = entry.getValue();
+
+                String host = cluster.getHost(slot.getNodeId());
+                AssignmentInfo ass = new AssignmentInfo(slot, sa.getTopologyId(), new HashSet<ExecutorDetails>(executors));
+                List<AssignmentInfo> executorList = hostAssignments.get(host);
+                if (executorList == null) {
+                    executorList = new ArrayList<AssignmentInfo>();
+                    hostAssignments.put(host, executorList);
+                }
+                executorList.add(ass);
+            }
+        }
+        return hostAssignments;
+    }
+
+    private Set<Set<ExecutorDetails>> computeWorkerSpecs(TopologyDetails topology) {
```

Comment:
It might be better to keep the value name to `details`, or use full name `topologyDetails`.